main.cpp

```diff
@@ -19,13 +19,28 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
+    CastCallBack(Rewriter& _rewriter):rewriter(_rewriter) {
         // Your code goes here
     };
 
     void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+        auto csce = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+	if((csce!= nullptr)&&(!csce->getExprLoc().isMacroID())){
+	auto  csce_gs = csce->getSubExprAsWritten()->IgnoreImpCasts();
+	auto var_end = Lexer::getLocForEndOfToken(csce_gs->getEndLoc(), 0, *Result.SourceManager, LangOptions());
+	auto var_start = csce->getSubExprAsWritten()->getBeginLoc();
+	auto l_brace = csce->getLParenLoc();
+	auto r_brace = csce->getRParenLoc();
+	rewriter.ReplaceText(l_brace,1,"static_cast<");
+	rewriter.ReplaceText(CharSourceRange::getCharRange(r_brace,var_start),">");

```

Comment:
Please descrbe the difference between the use of `getCharRange()` and `getTokenRange()` with code examples.

Reply:
The difference is:
getTokenRange() returns all tokens in range from the start to the start of  last token.
getCharRange() returns the range of characters from starting to ending by the specified locations.
we can see the diifference on test sample
with getCharRange::
`rewriter.ReplaceText(CharSourceRange::getCharRange(r_brace,var_start),">");`
`if(!isa<ParenExpr>(csce_gs)){`
    `// rewriter.InsertText(var_start, "(");`
    `// rewriter.InsertText(var_end, ")");`
`}`
result:
`int i = static_cast<int>fooooo ;`

with getTokenRange
`rewriter.ReplaceText(CharSourceRange::getTokenRange(r_brace,var_start),">");`
`if(!isa<ParenExpr>(csce_gs)){`
    `// rewriter.InsertText(var_start, "(");`
    `// rewriter.InsertText(var_end, ")");`
`}`
result:
`int i = static_cast<int> ;`